src/System.Management.Automation/engine/parser/AstVisitor.cs

```diff
@@ -175,8 +175,15 @@ public interface ICustomAstVisitor2 : ICustomAstVisitor
         object VisitDynamicKeywordStatement(DynamicKeywordStatementAst dynamicKeywordAst);
     }
 
+    /// <summary/>
+    public interface ICustomAstVisitor3 : ICustomAstVisitor2
```

Comment:
I think numbered interfaces made sense when the release cycle spanned years, but I'm worried we'll end up with `ICustomAstVistor9` before too long.

Reply:
One (maybe not awesome) idea is to avoid the problem for this feature is to add a `bool` flag to `ArrayLiteralAst` - because they should be structurally the same.

The downside to this approach is that some analysis tools that aren't aware of list literals will analyze the ast as an array, e.g. they may incorrectly give warnings about performance.

Reply:
I thought about reusing the `ArrayLiteralAst`, but didn't go that direction for two reasons:
1. An explicit `ListLiteralAst` is more discoverable.
2. `ArrayLiteralAst` can appear on the left side of an assignment. I didn't plan to make `[]` also assignable. (but I don't object the idea to make it able to do the same).

I would like to see more discussion on this before updating the code.